daemons/ipa-otpd/queue.c

```diff
@@ -111,6 +111,8 @@ void otpd_queue_push(struct otpd_queue *q, struct otpd_queue_item *item)
         q->head = q->tail = item;
     else
         q->tail = q->tail->next = item;
+
+    item->next = NULL;
 }
 
 void otpd_queue_push_head(struct otpd_queue *q, struct otpd_queue_item *item)
```

Comment:
In otpd_queue_push_head, we should also set item->next = NULL if the queue is currently empty.

Reply:
Yes it should; good catch!  Fixed.